src/mpi/romio/adio/ad_daos/ad_daos_features.c

```diff
@@ -0,0 +1,32 @@
+/* -*- Mode: C; c-basic-offset:4 ; indent-tabs-mode:nil ; -*- */
+/*
+ *
+ * Copyright (C) 2018-2019 Intel Corporation
+ *
+ * GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+ * The Government's rights to use, modify, reproduce, release, perform, display,
+ * or disclose this software are subject to the terms of the Apache License as
+ * provided in Contract No. 8F-30005.
+ * Any reproduction of computer software, computer software documentation, or
+ * portions thereof marked with this legend must also reproduce the markings.
+ */
+
+#include "adio.h"
+#include "ad_daos.h"
+
+int ADIOI_DAOS_Feature(ADIO_File fd, int flag)
+{
+    switch (flag) {
+        case ADIO_TWO_PHASE:
+        case ADIO_SCALABLE_OPEN:
```

Comment:
I thought you had a scalable open?  Isn't that what the inlined `handle_share` is doing?  I haven't gotten to the open code yet...

Reply:
i think i was copy-pasting here from somewhere.
I am not sure what setting this flag would mean here.

Reply:
im not sure what setting this flag would indicate TBH.
i think i was just copy-pasting here from somewhere

Reply:
Oh, yeah, I thought I was being more clever than I was.  'SCALABLE_OPEN' feature  is only checked at open time, and it disables deferred open -- since open is fast, there's no need to defer it  (https://github.com/pmodels/mpich/blob/master/src/mpi/romio/adio/common/ad_open.c#L143).   Not a huge deal but deferred open helps GPFS and I presume Lustre a lot.  It requires setting special hints, though, so I would be shocked if many (any?) users request that feature.

Reply:
so if i understand right we should just return 1 for SCALABLE_OPEN to avoid the deferred open (even if it's not used by many/any)?
would deferred open require all collective ops? not sure how that would work since mpi_file_open is collective.

Reply:
Updated the PR to return true for scalable open and resize.